astropy/table/column.py

```diff
@@ -131,25 +80,28 @@ def col_iter_str_vals(col):
 def col_copy(col):
     """
     This is a mixin-safe version of Column.copy() (with copy_data=True).
-
-    Warning: this function is subject to change or removal.
     """
     if isinstance(col, BaseColumn):
         return col.copy()
 
-    if hasattr(col, '_astropy_column_attrs'):
-        col_setattr(col, 'parent_table', None)  # Don't copy weakref to parent table
-    newcol = col.copy() if hasattr(col, 'copy') else deepcopy(col)
+    col.info.parent_table = None  # Don't copy weakref to parent table
```

Comment:
This seems risky: aren't you somehow detaching the column to be copied from its parent table here? Shouldn't one just avoid copying that particular link?